tests/core/test_device_simulation.py

```diff
@@ -0,0 +1,104 @@
+from typing import Hashable, Iterable, Mapping, cast
+
+import pytest
+from immutables import Map
+from mock import Mock, create_autospec, patch
+
+from tickit.core.adapter import AdapterConfig, ListeningAdapter
+from tickit.core.components.device_simulation import DeviceSimulation
+from tickit.core.state_interfaces.internal import (
+    InternalStateConsumer,
+    InternalStateProducer,
+)
+from tickit.core.typedefs import Changes, ComponentID, PortID, SimTime
+from tickit.devices.source import Source
+
+
+@pytest.fixture
+def source() -> Source:
+    source = Source(value=42)
+    return source
+
+
+@pytest.fixture
+def mock_adapter() -> Mock:
+    return create_autospec(AdapterConfig)
+
+
+@pytest.fixture
+def patch_asyncio() -> Iterable[Mock]:
+    with patch(
+        "tickit.core.components.device_simulation.asyncio", autospec=True
+    ) as mock:
+        yield mock
+
+
+@pytest.fixture
+def patch_internal_server() -> Iterable[Mock]:
+    with patch(
+        "tickit.core.state_interfaces.internal.InternalStateServer", autospec=True
+    ) as mock:
+        yield mock
+
+
+@pytest.fixture
+def device_simulation(
+    source: Source, mock_adapter: AdapterConfig, patch_internal_server
+) -> DeviceSimulation:
+    return DeviceSimulation(
+        name=ComponentID("test_device_simulation"),
+        state_consumer=InternalStateConsumer,
+        state_producer=InternalStateProducer,
+        device=source.SourceConfig(42),
+        adapters=[mock_adapter],
+    )
+
+
+def test_device_simulation_constructor(device_simulation: DeviceSimulation):
+    pass
+
+
+@pytest.mark.asyncio
+async def test_device_simulation_run_forever_method(
+    device_simulation: DeviceSimulation,
+    patch_asyncio: Mock,
+):
+    assert not hasattr(device_simulation, "state_producer")
+    assert not hasattr(device_simulation, "state_consumer")
+
+    await device_simulation.run_forever()
+
+    assert hasattr(device_simulation, "state_producer")
+    assert hasattr(device_simulation, "state_consumer")
+
+    assert isinstance(device_simulation.state_consumer, InternalStateConsumer)
+    assert isinstance(device_simulation.state_producer, InternalStateProducer)
```

Comment:
If we create a `test_set_up_state_interfaces_instantiates_state_interfaces`, we can replace this with an assertion it's awaited